### PR TITLE
[docs] Add precision about `pageExtensions`

### PR DESCRIPTION
**Pull request by yoannmoinet**
Adding a precision because when you change the `pageExtensions` configuration in `next.config.js`, `middleware.ts` isn't recognized anymore.

Took me some time to debug this one, not sure if it should be considered a bug or just lack of documentation.
I'll let you be the judge of that.

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

**Review by ijjk — COMMENTED** [review]
Hi, I'm not sure we need to document this here as we already explain this under the `pageExtensions` doc here https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions

**Comment by yoannmoinet** [comment]
@ijjk ah! I totally missed it.

I added the middleware way after I touched `pageExtensions` though, so I didn't go back to the `pageExtensions` doc when I had an issue with the middleware.

I'm fine closing this, it's just that I figured, if it happened to me, might happen to others.
Feel free to close this.

File: docs/advanced-features/middleware.md [hunk 1]
```diff
@@ -50,6 +50,9 @@ export const config = {
 }
 ```
 
+> **Note** If you're changing `pageExtensions` in `next.config.js` you'll have to update your middleware file name.
+> ex: With `pageExtensions: ['page.tsx', 'page.ts']` it will be `middleware.page.ts`.
```
Comment (ijjk) [inline docs/advanced-features/middleware.md:54]:
```suggestion
> **Note** the `pageExtensions` config affects middleware as well, [see related documentation here](https://nextjs.org/docs/api-reference/next.config.js/custom-page-extensions).
```

**Review by ijjk — APPROVED** [review]
Sounds good to add this note, updated to link to the existing `pageExtensions` doc. Thanks!

**Pull request merged**